LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,352 @@
+#pragma once
+#include <functional>
+
+template <typename Type>
+class TSLList
+{
+private:
+	int Size;
+	struct FNode
+	{
+		Type Element;
+		FNode* Next = nullptr;
+	};
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+
+public:
+	TSLList()
+	{
+		Size = 0;
+	}
+
+	~TSLList()
+	{
+		Clear();
+	}
+
+	//--------------------------------FIterator
+	class FIterator
+	{
+	private:
+		FNode* Current;
+
+	public:
+		FIterator(FNode* NewCurrent)
+		{
+			Current = NewCurrent;
+		}
+
+		FIterator& operator++()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		Type& operator*()
+		{
+			return Current->Element;
+		}
+
+		const Type& operator*() const
+		{
+			return Current->Element;
+		}
+
+		bool operator!=(const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+
+		bool operator!=(const FIterator& Other) const
+		{
+			return Current != Other.Current;
+		}
+	};
+
+	FIterator begin()
+	{
+		if(Head)
+			return FIterator(Head);
+		return FIterator(nullptr);
+	}
+
+	FIterator end()
+	{
+		return FIterator(nullptr);
+	}
+
+	//-------------------------------CopyConstructor
+	TSLList(const TSLList<Type>& OtherTSLlist)
+	{
+		FNode* Temporal = OtherTSLlist.Head;
```

Comment:
Hmm, I think this variable might be understood a lot better if it were named something like _Current_, for example. Temporal (which means related to time in English) / Temporary might be a bit confusing, as what it is pointing to is a more "permanent" node in the list as opposed to a temporary created one.